nomos-utils/src/bounded_duration.rs

```diff
@@ -0,0 +1,181 @@
+pub type MINUTE = BoundTag<'m'>;
+/// *Hour* tag type for `MinimalBoundedDuration`
+pub type HOUR = BoundTag<'h'>;
+/// *Day* tag type for `MinimalBoundedDuration`
+pub type DAY = BoundTag<'d'>;
+
+// we have a limitation to const types, so we better use the ones defined.
+fn fill_duration_measure<T: TimeTag>() -> Result<String, impl Display> {
+    match T::inner() {
+        v @ ('d' | 'h' | 'm' | 's') => Ok(v.to_string()),
+        'l' => Ok("ms".to_string()),
+        'n' => Ok("ns".to_string()),
+        other => Err(format!("'{other}' measure not supported")),
+    }
+}
+
+impl<const MIN_DURATION: usize, TAG: TimeTag> SerializeAs<Duration>
+    for MinimalBoundedDuration<MIN_DURATION, TAG>
+{
+    fn serialize_as<S>(source: &Duration, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        source.serialize(serializer)
+    }
+}
+
+impl<'de, const MIN_DURATION: usize, TAG: TimeTag> DeserializeAs<'de, Duration>
+    for MinimalBoundedDuration<MIN_DURATION, TAG>
+{
+    fn deserialize_as<D>(deserializer: D) -> Result<Duration, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let duration: MinimalBoundedDuration<MIN_DURATION, TAG> =
+            MinimalBoundedDuration::deserialize(deserializer)?;
+        Ok(duration.duration)
+    }
+}
+
+impl<'de, const MIN_DURATION: usize, TAG: TimeTag> Deserialize<'de>
+    for MinimalBoundedDuration<MIN_DURATION, TAG>
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Duration = Duration::deserialize(deserializer)?;
+        let reformat_measure = fill_duration_measure::<TAG>().map_err(Error::custom)?;
+        let parsed_duration =
+            humantime::parse_duration(format!("{MIN_DURATION}{reformat_measure}").as_str())
+                .map_err(Error::custom)?;
+        let min_duration: Duration = Duration::new(
+            parsed_duration
+                .as_secs()
+                .try_into()
+                .map_err(Error::custom)?,
+            parsed_duration
+                .subsec_nanos()
+                .try_into()
+                .map_err(Error::custom)?,
+        );
+        if value < min_duration {
+            return Err(Error::custom(format!(
+                "Minimal duration is {min_duration} but got {value}"
+            )));
+        }
+        Ok(Self {
+            duration: value,
+            _tag: PhantomData,
+        })
+    }
+}
+
+impl<const MIN_DURATION: usize, TAG: TimeTag> From<MinimalBoundedDuration<MIN_DURATION, TAG>>
+    for Duration
+{
+    fn from(value: MinimalBoundedDuration<MIN_DURATION, TAG>) -> Self {
+        value.duration
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use serde_with::serde_as;
+    use time::Duration;
+
+    use crate::bounded_duration::{MinimalBoundedDuration, DAY, SECOND};
+
+    #[test]
+    fn success_deserialize() {
+        let value = serde_json::to_value(Duration::days(1)).unwrap();
```

Comment:
I think it would be better to use a literal JSON string to give an idea of how it would look like when serialized.